core/mxapi.h

```diff
@@ -361,6 +361,14 @@ struct MxOps
    * - @a path is not NULL and not empty
    */
   int (*path_is_empty)     (const char *path);
+
+  /**
+   * mbox_create - create a new mailbox
+   * @param a Account to create the mailbox on
+   * @param path
+   * @param name The name of the mailbox
+  */
+  struct Mailbox *(*mbox_create)       (struct Account *a, const char *path, const char *name);
```

Comment:
> The Mailbox tells the Account where to create the new folder
>  (passing NULL would mean create a top-level folder)

Where do we get the path info in this case? Is there some path info in the Account object already? (at a glance I didn't see much useful `Account.adata` for maildir)

Reply:
Ah! You've got me there.
I'm thinking of my `account` command (not yet implemented).

We can probably use `$folder` as the root -- create `name` beneath it.
Hmm... but the backends shouldn't know about `$folder`.
I'll keep thinking.

Reply:
I think for now, we do need to pass the entire path (rather than the name) for Maildir.
We can ask the user for the name, then `if (m->type == MUTT_MAILDIR)` we can swap it for the full path.

Currently, the Maildir backend doesn't have its own AccountData.
I'll have a think about how it should be set up.